utils/common/src/main/kotlin/Extensions.kt

```diff
@@ -33,6 +33,11 @@ import java.nio.file.attribute.BasicFileAttributes
 import java.util.EnumSet
 import java.util.Locale
 
+/**
+ * Conveniently call [also] only if the receiver is null, e.g. for error handling, and return the receiver in any case.
+ */
+inline fun <T> T.alsoIfNull(block: (T) -> Unit): T = this ?: also(block)
```

Comment:
Should / can this have the `T?` ? `inline fun <T> T?.alsoIfNull`

Reply:
The code block is only executed if `T` is null, so the parameter could be
`() -> Unit` instead.

Reply:
> Should / can this have the `T?` ? `inline fun <T> T?.alsoIfNull`

No, in contrast to `Any.alsoIfNull` (which does not match `Any?`), using a generic `T` includes `T?`. FYI, I copied the function signature from Kotlin's `also` extension function.

Reply:
> The code block is only executed if `T` is null, so the parameter could be `() -> Unit` instead.

No, it couldn't, as I'm passing `block` on to `also`, which requires this function signature.

Reply:
> No, it couldn't, as I'm passing block on to also

Hm, I basically meant something like the following:
```
inline fun <T> T.alsoIfNull(block: () -> Unit): T = this ?: run {
    block()
    null
}
```
e.g. not calling `also` at all. What do you think ?

Reply:
I prefer calling `also` internally as it's slighly more compact and highlights the relation between these two functions.